### ISSUE TITLE
mpirun noticed that process rank 0 with PID 0 on node -MacBook-Air exited on signal 6 (Abort trap: 6).

### ISSUE THREAD

**Issue by lizat-i** [open-mpi/ompi#11595]
**## Background information**

Hello, I am running a simple MPI hello world program, which works on my Linux machine but gives me this warning 

**mpirun noticed that process rank 1 with PID 0 on node Ivans-MacBook-Air exited on signal 6 (Abort trap: 6).**

This is the full output : 

```
mpirun -np 2 a.out

Hello, MPI! Rank: 1 Total: 2 Machine: Ivans-MacBook-Air.local
Hello, MPI! Rank: 0 Total: 2 Machine: Ivans-MacBook-Air.local
 
Primary job  terminated normally, but 1 process returned
a non-zero exit code. Per user-direction, the job has been aborted.
 

**mpirun noticed that process rank 1 with PID 0 on node Ivans-MacBook-Air exited on signal 6 (Abort trap: 6).**
```

### What version of Open MPI are you using? 

mpirun (Open MPI) 4.1.5

### Describe how Open MPI was installed:

I tried two installation ways both resulted in the same behaviour:
1)     brew install openmpi
2)   download from https://www.open-mpi.org/software/ompi/v4.1/
openmpi-4.1.5.tar.gz	16.97 MiB	Feb 23, 2023	MD5: 2593008bea4bc721b9f304428abbf94b
SHA1: 7cf262ee4dac79178c988a14924232131f8bcb43
and make it 

### Please describe the system on which you are running

* Operating system/version:
* 
ProductName:            macOS
ProductVersion:         13.3.1
BuildVersion:           22E261

* Computer hardware:
 Darwin Ivans-MacBook-Air.local 22.4.0 Darwin Kernel Version 22.4.0: Mon Mar  6 21:01:02 PST 2023; RELEASE_ARM64_T8112 arm64
* Network type:

-----------------------------

## Details of the problem

Source Code: 
```c++
#include <mpi.h>
#include <iostream>

using namespace std;

int main(int argc, char *argv[]) {
  int rank;

  int size;

  char name[80];
 
  int length;

  MPI_Init(&argc, &argv);

 
  MPI_Comm_rank(MPI_COMM_WORLD, &rank);

 
  MPI_Comm_size(MPI_COMM_WORLD, &size);
 
  MPI_Get_processor_name(name, &length);
 
  cout << "Hello, MPI! Rank: " << rank << " Total: " << size
       << " Machine: " << name << endl;

 
  MPI_Finalize();

  return 0;
}
```

**compilation:** 

 mpic++ 5_mpi_standard.cpp    

**execution:**

mpirun -np 2 a.out

**Comment by jsquyres** [comment]
@lizat-i Your hostname array needs to be of (at least) length `MPI_MAX_PROCESSOR_NAME`, not 80.  See https://docs.open-mpi.org/en/v5.0.x/man-openmpi/man3/MPI_Get_processor_name.3.html.

**Comment by lizat-i** [comment]
Thank you! This solved the issue!

**Comment by jsquyres** [comment]
Excellent; glad this fixed the issue.

If you care, what was happening was that since the MPI standard requires that the string buffer be at least `MPI_MAX_PROCESSOR_NAME` length long (and in Open MPI `MPI_MAX_PROCESSOR_NAME` is 256), your buffer was too short.  Open MPI called `gethostname(3)` with your buffer, and told the OS that the buffer was 256 bytes long.  The OS therefore apparently wrote past the end of your buffer, and wrote to something that ultimately ended up causing an abort during Open MPI's finalization / shutdown.

In short: buffer overruns are bad.  😄

**Comment by Lee-zifei** [comment]
g b u!Thank u sir!
